### ISSUE TITLE
Problems with not being able to set shiftKeyboardStep for NumberControl.

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/scenery-phet#699]
In NumberControl:

```js
217    assert && assert( !options.sliderOptions.hasOwnProperty( 'shiftKeyboardStep' ), 'NumberControl sets shiftKeyboardStep' );
...
223      // pdom - shiftKeyboardStep should be the same as clicking the arrow buttons
224      shiftKeyboardStep: options.delta,
```

I understand the motivation for this. But I'm constantly bumping into this assertion, and I'm not convinced that this needs to be a "hard" requirement.  I think it would be nice to allow clients to optionally provide it, and default to `shiftKeyboardStep` if not provided. 

For Fourier, I've typically been provided with requirements like _"For the Center control, set keyboard step to PI/2, shift to PI/4, and page to PI."_   

So I translate that to:

```js
        keyboardStep: Math.PI / 2,
        shiftKeyboardStep: Math.PI / 4,
        pageKeyboardStep: Math.PI
```

... and then the `shiftKeyboardStep` assertion in NumberControl fails.

So I've started doing this, to remind my future self not to try to set `shiftKeyboardStep`:

```js
        keyboardStep: Math.PI / 2,
        // shiftKeyboardStep is set to options.delta by NumberControl
        pageKeyboardStep: Math.PI
```

So I think there are 3 problems with not allowing client to set `shiftKeyboardStep ` for NumberControl:

(1) If hides part of the specification for keyboard steps. I can't put it in the options, and I have to look inside NumberControl to figure it out.
(2) If makes the keyboard-step API different from Slider.
(3) Less importantly, it assumes that `shiftKeyboardStep` will always be the same as `delta`, which I'll bet we decide is not the case at some point in the future. 

Assigning to @zepumph (implementor) to reconsider.

**Comment by zepumph** [comment]
>If hides part of the specification for keyboard steps. I can't put it in the options, and I have to look inside NumberControl to figure it out.

My thoughts were that this actually simplified the API by removing the duplication needed to specify delta and shiftKeyboardStep. I'm happy to remove the assertion though. 

Over to @jessegreenberg to confirm this seems reasonable to him.

**Comment by zepumph** [comment]
@jessegreenberg please assign back to @pixelzoom once you have spot checked this issue.

**Comment by pixelzoom** [comment]
Raising review priority to high. This is going to publish with Fourier 1.0, which will be going into dev-test soon.

**Comment by jessegreenberg** [comment]
Change seems reasonable to me. To review I just quickly tested passing a custom `shiftKeyboardStep` in `sliderOptions` to a NumberControl to verify the option worked. Closing.

**Comment by jessegreenberg** [comment]
Oh, back to @pixelzoom from the request: 
> @jessegreenberg please assign back to @pixelzoom once you have spot checked this issue.

**Comment by pixelzoom** [comment]
👍🏻 closing.
